Demo/src/main/java/com/paypal/android/ui/selectcard/SelectCardFragment.kt

```diff
@@ -0,0 +1,172 @@
+        savedInstanceState: Bundle?
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            MaterialTheme {
+                Surface(modifier = Modifier.fillMaxSize()) {
+                    SelectCardView(onTestCardSelected = { onTestCardSelected(it) })
+                }
+            }
+        }
+    }
+
+    private fun onTestCardSelected(card: TestCard) {
+        navigateToCardForm(card)
+    }
+
+    private fun navigateToCardForm(card: TestCard? = null) {
+        findNavController().navigate(
+            SelectCardFragmentDirections.actionSelectCardFragmentToCardFragment(card)
+        )
+    }
+
+    @OptIn(ExperimentalFoundationApi::class)
+    @ExperimentalMaterial3Api
+    @Composable
+    fun SelectCardView(
+        viewModel: SelectCardViewModel = viewModel(),
+        onTestCardSelected: (TestCard) -> Unit = {}
+    ) {
+        LazyColumn(
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            stickyHeader {
+                TestCardHeader("Manual Card Entry")
+            }
+            item {
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .background(Color(color = 0xFF0079C1)) // PayPal blue
+                        .selectable(
+                            selected = false,
+                            onClick = { navigateToCardForm() }
+                        )
+                ) {
+                    Text(
+                        text = "ENTER CARD MANUALLY️",
+                        color = Color.White,
+                        style = MaterialTheme.typography.headlineSmall,
+                        modifier = Modifier
+                            .align(Alignment.Center)
+                            .padding(vertical = 26.dp, horizontal = 8.dp),
+                    )
+                }
+            }
+            stickyHeader {
+                TestCardHeader("Test Cards without 3DS")
+            }
+            items(viewModel.nonThreeDSCards) { card ->
+                TestCardView(
+                    testCard = card,
+                    selected = false,
+                    onClick = {
+                        onTestCardSelected(card)
+                    }
+                )
+                Divider(color = Color.Black)
+            }
+            stickyHeader {
+                TestCardHeader("Test Cards with 3DS")
+            }
+            items(viewModel.threeDSCards) { card ->
+                TestCardView(
+                    testCard = card,
+                    selected = false,
+                    onClick = {
+                        onTestCardSelected(card)
+                    }
+                )
+                Divider(color = Color.Black)
+            }
+        }
+    }
+
+    @ExperimentalMaterial3Api
+    @Preview
```

Comment:
How did Android Studio's preview tool work for you throughout this development? The Xcode one for SwiftUI was originally really buggy but _apparently_ is relatively stable now

Reply:
It's ok, I'll say it's better than rebuilding every time to see changes, but that's about it. They're still really early on with Compose too, so they'll probably keep improving it over time. It definitely bugs out sometimes though.

Reply:
During my sabbatical I used the Swift UI previews a bit and it's way more stable/easier to develop with than it was initially! It's really easy to interact with as well and go through screens all in the previews. 🤓